File: ProgramacionJava/Ejercicios Navidad/src/banco/TarjetaPrepago.java
package banco;
import dnicif.DniCif;
import fechaHora.Hora;


public class TarjetaPrepago {
	private String telefono;
	private DniCif dni;
	private Double saldo;
	private Hora consumo;

	public TarjetaPrepago(){}

	public TarjetaPrepago(String telefono, Double saldo,DniCif dni, Hora consumo){
		this.telefono	= telefono;
		this.dni		= dni;
		this.saldo		= saldo;	
		this.consumo	= consumo;
	}


	public String getTelefono() {
		return telefono;
	}

	public void setTelefono(String telefono) {
		this.telefono = telefono;
	}

	public DniCif getDni() {
		return dni;
	}

	public void setDni(DniCif dni) {
		this.dni = dni;
	}

	public Double getSaldo() {
		return saldo;
	}

	public void setSaldo(Double saldo) {
		this.saldo = saldo;
	}

	public Hora getConsumo() {
		return consumo;
	}

	public void setConsumo(Hora consumo) {
		this.consumo = consumo;
	}


	void enviarMensaje(Integer mensajes){
		this.saldo =- 0.09 * mensajes;
	}
	
	void realizarLlamada(Integer segundos){
		this.saldo =- 0.15 + (0.01 * segundos);
		this.contadorConsumo(segundos);
	}
	
	void ingresarSaldo(Double cantidad){
		this.saldo =+ cantidad;
	}
	
	
	void consultarTarjeta(){
		System.out.println("Telefono --> " + this.getTelefono());
		System.out.println("Saldo Tarjeta --> " + this.getSaldo());
	}
	
	void contadorConsumo(Integer segundosLlamada){
		Integer horas		= consumo.getHora();
		Integer minutos		= consumo.getMinutos();
		Integer segundos	= consumo.getSegundos() + segundosLlamada;
		
		if (segundos >= 60){
			minutos += segundos /60;
			
		if (minutos >= 60){
			horas += minutos /60;
		
		if (horas < 0)
			horas = 0;
		}
		}
		
		if (minutos >= 60){
			horas += minutos /60;
		
		if (horas < 0)
			horas = 0;
		}
		
		this.consumo.setHora(horas, minutos, segundos);
	}
}

File: ProgramacionPython/POO/TarjetaPrepago.py
class TarjetaPrepago():
        def __init__(self, numeroTelefono, nif, saldo):
                self.numeroTelefono = numeroTelefono
                self.dni            = dni
                self.saldo          = ""
                self.consumo        = 0
                #self.consumo        = Hora()



#Setters y Getters

        def setNumeroTelefono(self, numeroTelefono):
                self.numeroTelefono = numeroTelefono

        def getNumeroTelefono(self):
                return self.numeroTelefono

        def setNif(self, nif):
                self.nif = nif

        def getNif(self):
                return self.nif

        def setSaldo(self, saldo):
                self.saldo = saldo

        def getSaldo(self):
                return self.saldo

        def setConsumo(self, consumo):
                self.consumo = consumo

        def getConsumo(self):
                return self.consumo

        #Logica

        def ingresarSaldo(self, cantidad):
                self.saldo = saldo + cantidad

        def enviarMensaje(self, numeroMensajes):
                self.saldo = (saldo - (numeroMensajes * 9))

        def realizarLlamada(self, segundosHablados):
                self.saldo = (saldo - (15 + 1 * segundosHablados))
                                              
        def consultarTarjeta(self):
                print ("Numero--> ", getNumeroTelefono(), "\nDNI--> ", getNumeroTelefono(), "\nSaldo--> ", getSaldo(), "\nConsumo--> ", getConsumo())

        #Casos Test
        if __name__ == '__main__':
                        tarjeta = TarjetaPrepago()
                        tarjeta.numeroTelefono("697968370")
                        tarjeta.nif("43230867K")
                        tarjeta.saldo("50")
                        tarjeta.consumo(30)

                        print (consultarTarjeta())
        from Hora import *

        class TarjetaPrepago():
                def __init__(self, numeroTelefono, nif, saldo):
                        self.numeroTelefono = ""
                        self.nif            = ""
                        self.saldo          = ""
                        self.consumo        = 0
                        self.consumo        = Hora()



        #Setters y Getters

                def setNumeroTelefono(self, numeroTelefono):
                        self.numeroTelefono = numeroTelefono

                def getNumeroTelefono(self):
                        return self.numeroTelefono

                def setNif(self, nif):
                        self.nif = nif

                def getNif(self):
                        return self.nif

                def setSaldo(self, saldo):
                        self.saldo = saldo

                def getSaldo(self):
                        return self.saldo

                def setConsumo(self, consumo):
                        self.consumo = consumo

                def getConsumo(self):
                        return self.consumo

        #Logica

                def ingresarSaldo(self, cantidad):
                        self.saldo = saldo + cantidad

                def enviarMensaje(self, numeroMensajes):
                        self.saldo = (saldo - (numeroMensajes * 9))

                def realizarLlamada(self, segundosHablados):
                        self.saldo = (saldo - (15 + 1 * segundosHablados))
                                      
                def consultarTarjeta(self):
                        print ("Numero--> ", getNumeroTelefono(), "\nDNI--> ", getNumeroTelefono(), "\nSaldo--> ", getSaldo(), "\nConsumo--> ", getConsumo())

        #Casos Test
        if __name__ == '__main__':

                        tarjeta = TarjetaPrepago()
                        tarjeta.numeroTelefono("697968370")
                        tarjeta.nif("43230867K")
                        tarjeta.saldo("50")
                        tarjeta.consumo(30)

                        print (consultarTarjeta())
